resources/schema.py

```diff
@@ -328,6 +340,121 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         return DeleteBerthTypeMutation()
 
 
+class HarborInput(AbstractAreaInput):
+    municipality_id = graphene.String()
+    image_file = graphene.String()
+    availability_level_id = graphene.ID()
+    number_of_places = graphene.Int()
+    maximum_width = graphene.Int()
+    maximum_length = graphene.Int()
+    maximum_depth = graphene.Int()
+    name = graphene.String()
+    street_address = graphene.String()
+
+
+class CreateHarborMutation(graphene.ClientIDMutation):
+    class Input(HarborInput):
+        pass
+
+    harbor = graphene.Field(HarborNode)
+
+    @classmethod
+    @login_required
+    @superuser_required
+    @transaction.atomic
+    def mutate_and_get_payload(cls, root, info, **kwargs):
+        # TODO: Should check if the user has permissions to
+        # delete the specific resource
+        lang = get_language()
+
+        availability_level_id = kwargs.pop("availability_level_id", None)
+        if availability_level_id:
+            try:
+                kwargs["availability_level"] = AvailabilityLevel.objects.get(
+                    pk=availability_level_id
+                )
+            except AvailabilityLevel.DoesNotExist as e:
+                raise VenepaikkaGraphQLError(e)
+
+        municipality_id = kwargs.pop("municipality_id", None)
+        if municipality_id:
+            try:
+                kwargs["municipality"] = Municipality.objects.get(id=municipality_id)
+            except Municipality.DoesNotExist as e:
+                raise VenepaikkaGraphQLError(e)
+
+        try:
+            harbor = Harbor.objects.language(lang).create(**kwargs)
+        except Exception as e:
```

Comment:
can we maybe try to catch some more specific errors? IntegrityError, I guess is one.

Reply:
Yeah, can also do that, I was aiming to catch anything unexpected, but we could either (1) catch specific errors, like Integrity, or just let it bubble in case it would happen.

Reply:
We're not really catching such errors anywhere else, I was just thinking it could be useful, _just in case_.

Reply:
the reason being, `VenePaikkaError` would not be sent to sentry, other errors will, once we implement a new `GraphQLSentryView`

Reply:
Let's remove it then, otherwise we'll be in the same situation where the `IntegrityError` is absorbed by the `transaction.atomic`.